File: src/FrameLog/Models/LogModels.cs
﻿using System;
using static FrameHelper.EntityHelpers;

namespace FrameLog.Models
{
    public interface ILogDTO : IWithIdAndTimeStamp
    {
        Guid? UserId { get; set; }
        string Type { get; set; }
        string Message { get; set; }
        string Stack { get; set; }
        string Location { get; set; }
    }

    public class LogDTO : ILogDTO
    {
        public Guid Id { get; set; }
        public DateTime TimeStamp { get; set; }
        public Guid? UserId { get; set; }
        public string Type { get; set; }
        public string Message { get; set; }
        public string Stack { get; set; }
        public string Location { get; set; }
    }

    public interface ILogView : ILogDTO
    {
    }

    public class LogView : LogDTO, ILogView
    {
    }

}

File: src/FrameUI/src/app/models/LogModels.ts
export interface IlogDTO {
    id: string;
    timeStamp: Date;
    userId: string;
    type: string;
    message: string;
    stack: string;
    location: string;
    
}

export class LogDTO implements IlogDTO {
    public id: string;
    public timeStamp: Date;
    public userId: string;
    public type: string;
    public message: string;
    public stack: string;
    public location: string;
    
    constructor(model?: IlogDTO) {
        
        if(model) {
            this.id = model.id;
            this.timeStamp = model.timeStamp;
            this.userId = model.userId;
            this.type = model.type;
            this.message = model.message;
            this.stack = model.stack;
            this.location = model.location;
            
        }
    }
}
export interface IlogView extends IlogDTO {
    
}

export class LogView extends LogDTO implements IlogView {
    
    constructor(model?: IlogView) {
        super(model);
        if(model) {
            
        }
    }
}
 